### PR TITLE
ci: -Wl,--s390-pgste for s390x

### PR DESCRIPTION
**Pull request by Jakob-Naucke**
in static tests. Required for basic KVM checks on some kernels (e.g. the
one RHEL is currently shipping), cf.
https://github.com/qemu/qemu/blob/6621441db50d5bae7e34dbd04bf3c57a27a71b32/target/s390x/kvm/meson.build#L15-L16.

Fixes: #4375
Signed-off-by: Jakob Naucke <jakob.naucke@ibm.com>

Related: https://github.com/kata-containers/kata-containers/pull/3470

### CODE REVIEW

File: .ci/go-test.sh
Comment (devimc) [inline .ci/go-test.sh:247]: can `ldflags` be empty ? :thinking:

Comment (Jakob-Naucke) [reply .ci/go-test.sh:247]: yes, you can build with `-ldflags ''` without any consequences AFAICT (we will know with the tests for sure)

**Review by dborquez — APPROVED** [review]
thanks @Jakob-Naucke LGTM

File: .ci/go-test.sh
Comment (wainersm) [inline .ci/go-test.sh:244]: Nit: on https://github.com/kata-containers/tests/tree/main/.ci/go-test.sh#L26 it uses `"$(go env GOARCH)"` to get the arch. In case you want to keep it consistent.

Yet-another-Nit: do you need to pass `-Wl`? If I am not wrong, in C, `-Wl,` is used to pass a linker argument via compiler. Given that you are passing it direct to the linker via `-extldflags` maybe you don't need it. Not a golang expert speaking...

Comment (Jakob-Naucke) [reply .ci/go-test.sh:244]: > Nit: on https://github.com/kata-containers/tests/tree/main/.ci/go-test.sh#L26 it uses `"$(go env GOARCH)"` to get the arch. In case you want to keep it consistent.

Consistency is another can of worms :) but okay, let me update this one

> Yet-another-Nit: do you need to pass `-Wl`? If I am not wrong, in C, `-Wl,` is used to pass a linker argument via compiler. Given that you are passing it direct to the linker via `-extldflags` maybe you don't need it. Not a golang expert speaking...

`-extldflags` is, slightly counterintuitively, passed to GCC: `s390x-linux-gnu-gcc: error: unrecognized command line option ‘--s390-pgste’`

Comment (wainersm) [reply .ci/go-test.sh]: > > Nit: on https://github.com/kata-containers/tests/tree/main/.ci/go-test.sh#L26 it uses `"$(go env GOARCH)"` to get the arch. In case you want to keep it consistent.
> 
> Consistency is another can of worms :) but okay, let me update this one
> 

hehehe unfortunately it is true

> > Yet-another-Nit: do you need to pass `-Wl`? If I am not wrong, in C, `-Wl,` is used to pass a linker argument via compiler. Given that you are passing it direct to the linker via `-extldflags` maybe you don't need it. Not a golang expert speaking...
> 
> `-extldflags` is, slightly counterintuitively, passed to GCC: `s390x-linux-gnu-gcc: error: unrecognized command line option ‘--s390-pgste’`

hmm... interesting. Thanks for checking that.

**Review by wainersm — APPROVED** [review]

File: .ci/go-test.sh
Comment (Jakob-Naucke) [reply .ci/go-test.sh]: we do need to escape because of `eval` though (updated)

**Review by jodh-intel — APPROVED** [review]
Thanks @Jakob-Naucke.

lgtm

**Review by fidencio — APPROVED** [review]
lgtm, thanks @Jakob-Naucke!

**Comment by fidencio** [comment]
A head's up that may affect your PR.

In the Architecture Committee meeting from January 25th, 2022, the Architecture Committee has agreed on using the "Dismiss stale pull request approvals when new commits are pushed" configuration from GitHub.  It basically means that if your PR has been rebased or updated, the approvals given will be erased.

In order to minimize traumas due to the new approach, please, consider adding a note on the changes done before the rebase / force-push, and also pinging the reviewers for a subsequent round of reviews.

Thanks for your understanding!

Related issue: https://github.com/kata-containers/community/issues/249

**Comment by Jakob-Naucke** [comment]
@fidencio sorry, that still needed an update; shell quoting is the absolute worst, and here goes /test

**Comment by Jakob-Naucke** [comment]
@dborquez @fidencio @jodh-intel @wainersm if I may get a re-approval :)

**Review by fidencio — APPROVED** [review]
lgtm, thanks @Jakob-Naucke!

**Review by jodh-intel — APPROVED** [review]
Thanks @Jakob-Naucke.

lgtm

**Pull request merged**
